rclpy/src/rclpy/timer.cpp

```diff
@@ -37,20 +37,14 @@ void
 Timer::destroy()
 {
   rcl_timer_.reset();
-  clock_handle_.reset();
+  clock_.destroy();
+  context_.destroy_when_not_in_use();
```

Comment:
I think `context_.destroy()` is safe to use here since the `Timer` owns a copy, but it probably doesn't change anything.